tests/Integration/SearchIndexTest.php

```diff
@@ -0,0 +1,65 @@
+<?php
+
+namespace Algolia\AlgoliaSearch\Helper\Tests\Integration;
+
+use Algolia\AlgoliaSearch\Helper\Tests\Helpers\Factory;
+use PHPUnit\Framework\TestCase;
+
+final class SearchIndexTest extends TestCase
+{
+    /**
+     * @var \Algolia\AlgoliaSearch\Helper\SearchIndex
+     */
+    private $index;
+
+    /**
+     * @return void
+     */
+    public function setUp()
+    {
+        $client = Factory::getClient();
+
+        $this->index = $client->initIndex(Factory::getIndexName($this->getName()));
+    }
+
+    /**
+     * @return void
+     */
+    public function tearDown()
+    {
+        $this->index->delete();
+    }
+
+    /**
+     * @return void
+     */
+    public function testIndexExists()
```

Comment:
add the `testIndexDoesNotExist` method

Reply:
Already refactor with @nunomaduro, to much method, that's why we directly test if the indexDoesNotExist and after test if the index exist in the same method.